manifests/cron.pp

```diff
@@ -14,17 +18,27 @@
   }
 
   if $mailto != undef {
-    cron { 'aide':
-      ensure  => $cron_ensure,
-      command => "${aide_path} --check | /bin/mail -s \"\$(hostname) - AIDE Integrity Check\" ${mailto}",
-      user    => 'root',
-      hour    => $hour,
-      minute  => $minute,
+    if $mail_only_on_changes {
+      cron { 'aide':
+        ensure  => $cron_ensure,
+        command => "LOG=$(mktemp) && ${aide_path} --config ${conf_path} --check > \$LOG 2>&1 || ${cat_path} -v \$LOG | /usr/bin/mail -E -s \"\$(hostname) - AIDE Integrity Check\" ${mailto}; ${rm_path} -f \$LOG",
```

Comment:
Not only does this look pretty nasty but it's also buggy.

`/usr/bin/mail`

This is a different binary on different OSes and the `-E` means different things.
On Redhat, `/usr/bin/mail` is `mailx`? and `-E` means...

> If  an outgoing message does not contain any text in its first or only message part, do not send it but discard it silently, effectively setting the skipemptybody variable at program startup.
              This is useful for sending messages from scripts started by cron(8).

But on Ubuntu...

`/usr/bin/mail` is `mail.mailutils` and `-E` means something completely different...

> -E, --exec=COMMAND
              execute COMMAND

Reply:
Please provide me with more insights as to why is it nasty? and what is your recommendation on how to fix it? What you referenced for Ubuntu is: 
> Execute command before opening the mailbox. Any number of --exec options can be given. The commands will be executed after sourcing configuration files (see Mail Configuration Files), but before opening the mailbox.

Reply:
@alexjfisher Please provide me with more insights as to why is it nasty? and what is your recommendation on how to fix it? What you referenced for Ubuntu is: 
> Execute command before opening the mailbox. Any number of --exec options can be given. The commands will be executed after sourcing configuration files (see Mail Configuration Files), but before opening the mailbox.

Reply:
@alexjfisher I think I follow you now. ```/usr/bin/mail``` is ```mail.mailutils``` will only be true for Ubuntu 14 (which by the way end of life will be next month)

Reply:
On my Ubuntu 18.04 too... (it’s actually a symlink to `/etc/alternatives/mail`)

Reply:
> Execute command before opening the mailbox. Any number of --exec options can be given. The commands will be executed after sourcing configuration files (see Mail Configuration Files), but before opening the mailbox.

That's not what's intended though? In https://github.com/tenajsystems/puppet-aide/pull/1 I think the user's `/usr/bin/mail` must have been a link to `/usr/bin/mailx`.  That's available on Ubuntu too. Maybe use it directly (as there's very little guarantee as to what binary `/usr/bin/mail` actually uses).

> Please provide me with more insights as to why is it nasty?

Sorry.  That wasn't a good choice of phrasing.  Apologies.  I do think it might be fragile to have such a long cron command.  IMO at a certain point it's probably better to call a script.  I'm also unconvinced by parameterising the path to `cat` and `rm`.  I don't think there are any systems where `/bin/cat` and `/bin/rm` wouldn't work.
(honestly `cat` can probably be [avoided](http://porkmail.org/era/unix/award.html#cat) altogether.)

Reply:
Thank you for the detailed  information. I will work on fixing it.

Reply:
> honestly `cat` can probably be [avoided](http://porkmail.org/era/unix/award.html#cat) altogether.

`cat -v` was added in order to include non-printable characters in the email. File names can contain such characters.

Reply:
Ah, ok.  That makes sense.  *normally* you wouldn't expect to have such filenames, but since AIDE is specifically a security tool and looking for files that someone might be trying to hide from casual observers, this does make sense.

> -v, --show-nonprinting
              use ^ and M- notation, except for LFD and TAB